backend/layers/processing/schema_migration.py

```diff
@@ -102,16 +102,28 @@ def dataset_migrate(
         }
 
     def collection_migrate(self, collection_id: str, collection_version_id: str, can_publish: bool) -> Dict[str, Any]:
+        # Get datasets from collection
+        version = self.business_logic.get_collection_version(CollectionVersionId(collection_version_id))
+        current_schema_version = cellxgene_schema.schema.get_current_schema_version()
+
+        if not any(dataset.metadata.schema_version != current_schema_version for dataset in version.datasets):
+            # check if there are datasets to migrate.
+            return {
+                "can_publish": str(False),  # skip publishing, because the collection is already published and no
+                # revision is created, or the collection is private or a revision.
+                "collection_version_id": collection_version_id,
+                "datasets": [],
+                "no_datasets": str(True),
+            }
 
         if can_publish:
```

Comment:
edge case to consider: if a collection was originally a can_publish==True during run 1, but for whatever reason did not finish publishing perhaps 1 dataset of N failed to process or timed out), it would become a can_publish==False during run 2 because its in an open revision now. However, we can publish revisions that we created, just not those created before run 1. As an optimization, we should consider addressing this by either looking at timestamps or perhaps even looking for the machine account user under curator_name

Reply:
Up to you whether this should be a separate issue or you want to tackle it here

Reply:
Or we check if published in `publish_and_cleanup`. I'll add it to the PR and ask for another review.

Reply:
Will that account for the collections that were can_publish==False in run 1 and should stay can_publish==False in all subsequent runs? Just making sure we don't accidentally publish any private or revisions from before migration that are not ready

Reply:
> looking at timestamps

The time stamps sounds too variable to be reliable.

> perhaps even looking for the machine account user under curator_name

This is not reliable since most of the dataset will eventually be migrated by the machine account. Also are we setting the curator_name to the machine account when we migrate?

I think the best solution is to do it manually for now.